### DISCUSSION TITLE
Erroring out on Netbox Upgrade

### DISCUSSION THREAD

**Discussion by Cjericho4** [netbox-community/netbox#14020] [Q&A] [answered]
Hi all, 

I have been trying to upgrade our Netbox installation in Kubernetes to the latest version and am having some issues. 
We are running Postgres 14.5, and were previously running Netbox 3.2.8. We are trying to upgrade to the latest version of Netbox 3.6.3.

I am getting this error on the Netbox UI page: 
`<class 'django.db.utils.ProgrammingError'>

column extras_customlink.content_type_id does not exist
LINE 1: ...k"."created", "extras_customlink"."last_updated", "extras_cu...

Python version: 3.10.4
Netbox version: 3.2.8`

But when I check the pod for the error it gives me on the migration on startup it gives me this error:
`django.db.utils.IntegrityError: could not create unique index "dcim_device_unique_name_site_tenant"
DETAIL:  Key (lower(name::text), site_id, tenant_id)=(jmeterp, 2, 1) is duplicated.`

I have tried to roll back to the version of Netbox we were using and have been getting problems getting it started back up again. 

Anyone have any ideas on how to fix either the migration error to get it started?

**Comment by mtinberg** [comment, marked as answer]
My guess based on the error detail you included is that you have two Device definitions that only differ by case of the Name primary key which became case-insensitive in 3.5 or 3.6 (read the release notes), so you probably need to consolidate some records and delete the duplicate.  I'd say if Netbox doesn't start, either you can manually find and delete the record using psql and re-run the migrations or restore the db backup you made before you started (you have backups, right? 😉), edit in the WebUI on 3.2.x, then​ try to upgrade.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: CJ Bradshaw ***@***.***>
Sent: Wednesday, October 11, 2023 11:42 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Erroring out on Netbox Upgrade (Discussion #14020)


Hi all,

I have been trying to upgrade our Netbox installation in Kubernetes to the latest version and am having some issues.
We are running Postgres 14.5, and were previously running Netbox 3.2.8. We are trying to upgrade to the latest version of Netbox 3.6.3.

I am getting this error on the Netbox UI page:
`<class 'django.db.utils.ProgrammingError'>

column extras_customlink.content_type_id does not exist
LINE 1: ...k"."created", "extras_customlink"."last_updated", "extras_cu...

Python version: 3.10.4
Netbox version: 3.2.8`

But when I check the pod for the error it gives me on the migration on startup it gives me this error:
django.db.utils.IntegrityError: could not create unique index "dcim_device_unique_name_site_tenant" DETAIL: Key (lower(name::text), site_id, tenant_id)=(jmeterp, 2, 1) is duplicated.

I have tried to roll back to the version of Netbox we were using and have been getting problems getting it started back up again.

Anyone have any ideas on how to fix either the migration error to get it started?

**Comment by Cjericho4** [comment]
Thank you very much! Missed that section in the release notes and I really appreciate you catching that for me! 

Removed the duplicate entries and it was up and running very quickly!
